File: src/leetcode/3498reverseDegree.cpp
#include <iostream>
#include <string>

using namespace std;

class Solution {
public:
    int reverseDegree(string s) {
        int i = 1, sum = 0;
        for (char ch : s) {
            sum += (int(ch) - (71 + (int(ch) - 97) * 2)) * i;
            ++i;
        }
        return sum;
    }
};

int main(int argc, char const *argv[])
{
    string s = "abc";
    cout << Solution().reverseDegree(s) << endl;
    return 0;
}

File: scripts/3498reverseDegree.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        sum = 0
        for i, ch in enumerate(s):
            sum += (ord(ch) - (71 + 2 * (ord(ch) - ord("a")))) * (i + 1)
        return sum

s = "abc"
print(f"results: {Solution().reverseDegree(s)}")